extensions/builder.md

```diff
@@ -0,0 +1,145 @@
+
+## Environment Variables/Labels
+A builder's environment is the build-time environment of the stack, and as such, MUST fulfill all of the [build image specifications][build-image-specs].
+
+The following variables MUST be set in the builder environment (through the image config's `Env` field):
+
+| Env Variable    | Description
+|-----------------|--------------------------------------
+| `WorkingDir`  | The working directory where buildpacks should default (eg: `/workspace`)
+| `CNB_APP_DIR`  | Application directory of the build environment (eg: `/app`)
+| `CNB_LAYERS_DIR`  | The directory to create and store `layers` in the build environment (eg: `/layers`)
+| `CNB_PLATFORM_DIR`  | The directory to create and store platform focused files in the build environment (eg: `/platform`)
+
+The following variables MAY be set in the builder environment (through the image config's `Env` field):
+
+| Env Variable    | Description
+|-----------------|--------------------------------------
+| `SERVICE_BINDING_ROOT`  | The directory where services are bound
+
+The following labels MUST be set in the builder environment (through the image config's `Labels` field):
+
+| Env Variable    | Description
+|-----------------|--------------------------------------
+| `io.buildpacks.builder.api`  | The Builder API this builder implements (at this point, **0.1**)
+| `io.buildpacks.builder.metadata`  | A JSON object representing [Builder Metadata](#iobuildpacksbuildermetadata)
+| `io.buildpacks.buildpack.order`  | A JSON object representing the [order of the buildpacks stored on the builder](#iobuildpacksbuildpackorder)
+| `io.buildpacks.buildpack.layers`  | A JSON object representing the [buildpack layers](#iobuildpacksbuildpacklayers)
+| `io.buildpacks.lifecycle.version`  | A string, representing the version of the lifecycle stored in the builder (greater than `0.9.0`)
+| `io.buildpacks.lifecycle.apis`  | A JSON object representing the [lifecycle APIs](#iobuildpackslifecycleapis) the lifecycle stored in the builder supports
+
+### `io.buildpacks.builder.metadata`
+The `io.buildpacks.builder.metadata` data should look like:
+```json
+{
+  "description": "<Some description>",
+  "stack": {
+    "runImage": {
+      "image": "<some/run-image>",
+      "mirrors": [
+        "<gcr.io/some/default>"
+      ]
+    }
+  },
+  "buildpacks": [
+    {
+      "id": "<buildpack id>",
+	  "version": "<buildpack version>",
+	  "homepage": "http://geocities.com/top-bp"
+	}
+  ],
+  "createdBy": {
+    "name": "<creator of builder>",
+    "version": "<version of tool used to create builder>"
+  }
+}
+```
+
+### `io.buildpacks.buildpack.order`
+The `io.buildpacks.buildpack.order` data MUST look like:
+```json
+[
+	{
+	  "group":
+		[
+		  {
+			"id": "<buildpack id>",
+			"version": "<buildpack version>",
+			"optional": "<bool>"
+		  }
+		]
+	}
+]
+```
+
+### `io.buildpacks.buildpack.layers`
+The `io.buildpacks.buildpack.layers` data should look like:
```

Comment:
As above, should I mention that this could be an empty object, if there are no buildpacks stored on the builder?

Reply:
See https://github.com/buildpacks/spec/pull/193#discussion_r610105077